core/src/main/scala/org/http4s/rho/Router.scala

```diff
@@ -28,63 +30,63 @@ sealed trait RoutingEntity[T <: HList] {
   * @param rules header validation stack
   * @tparam T cumulative type of the required method for executing the router
   */
-case class Router[T <: HList](method: Method,
-                              path: PathRule,
-                              rules: RequestRule)
-  extends RouteExecutable[T]
-    with HeaderAppendable[T]
-    with RoutingEntity[T]
-    with Decodable[T, Nothing]
-    with RoutePrependable[Router[T]]
+case class Router[F[_], T <: HList](method: Method,
+                                    path: PathRule,
+                                    rules: RequestRule[F])
+  extends RouteExecutable[F, T]
+    with HeaderAppendable[F, T]
+    with RoutingEntity[F, T]
+    with Decodable[F, T, Nothing]
+    with RoutePrependable[F, Router[F, T]]
 {
-  type Self = Router[T]
+  type Self = Router[F, T]
 
-  override type HeaderAppendResult[T <: HList] = Router[T]
+  override type HeaderAppendResult[T <: HList] = Router[F, T]
 
-  override def /:(prefix: TypedPath[HNil]): Self = {
+  override def /:(prefix: TypedPath[F, HNil]): Self = {
     copy(path = PathAnd(prefix.rule, path))
   }
 
-  override def >>>[T1 <: HList](v: TypedHeader[T1])(implicit prep1: Prepend[T1, T]): Router[prep1.Out] =
+  override def >>>[T1 <: HList](v: TypedHeader[F, T1])(implicit prep1: Prepend[T1, T]): Router[F, prep1.Out] =
     Router(method, path, AndRule(rules, v.rule))
 
-  override def makeRoute(action: Action[T]): RhoRoute[T] = RhoRoute(this, action)
+  override def makeRoute(action: Action[F, T]): RhoRoute[F, T] = RhoRoute(this, action)
 
-  override def decoding[R](decoder: EntityDecoder[R])(implicit t: TypeTag[R]): CodecRouter[T, R] =
+  override def decoding[R](decoder: EntityDecoder[F, R])(implicit F: Functor[F], t: TypeTag[R]): CodecRouter[F, T, R] =
```

Comment:
bound here seem unused

Reply:
another case that comes from `http4s`, a bit more indirectly - it is necessary in line 90 for the `orElse`